services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/UpdateServerSettingsFragment.java

```diff
@@ -0,0 +1,152 @@
+
+public class UpdateServerSettingsFragment extends Fragment {
+
+    private class OnButtonClick implements View.OnClickListener {
+        @Override
+        public void onClick(View v) {
+            if (v.getId() == R.id.btnUpdateServerUrl) {
+                onClickUpdateUrl();
+            } else if (v.getId() == R.id.btnChooseDefaultServer) {
+                inputServerUrl.getEditText().setText(getString(R.string.default_sync_server_url));
+            } else if (v.getId() == R.id.btnVerifyServerUpdateServerDetails) {
+                startVerifyActivity();
+            } else if (v.getId() == R.id.btnScanQrUpdateServerDetails) {
+                // TODO
+            }
+        }
+    }
+
+    private TextInputLayout inputServerUrl;
+    private AbsSyncViewModel absSyncViewModel;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_update_server_settings, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        findViewsAndAttachListeners(view);
+        setupViewModelAndNavController();
+    }
+
+    private void findViewsAndAttachListeners(View view) {
+        inputServerUrl = view.findViewById(R.id.inputServerUrl);
+        Button btnUpdateUrl = view.findViewById(R.id.btnUpdateServerUrl);
+        Button btnSetDefault = view.findViewById(R.id.btnChooseDefaultServer);
+        Button btnVerifyServerDetails = view.findViewById(R.id.btnVerifyServerUpdateServerDetails);
+        Button btnScanQr = view.findViewById(R.id.btnScanQrUpdateServerDetails);
+
+        OnButtonClick onButtonClick = new OnButtonClick();
+        btnUpdateUrl.setOnClickListener(onButtonClick);
+        btnSetDefault.setOnClickListener(onButtonClick);
+        btnVerifyServerDetails.setOnClickListener(onButtonClick);
+        btnScanQr.setOnClickListener(onButtonClick);
+    }
+
+    private void setupViewModelAndNavController() {
+        absSyncViewModel = new ViewModelProvider(requireActivity()).get(AbsSyncViewModel.class);
+
+        absSyncViewModel.getServerUrl().observe(getViewLifecycleOwner(), s -> inputServerUrl.getEditText().setText(s));
+    }
+
+    private void startVerifyActivity() {
+        Intent intent = new Intent(requireActivity(), VerifyServerSettingsActivity.class);
+        intent.putExtra(IntentConsts.INTENT_KEY_APP_NAME, absSyncViewModel.getAppName());
+        startActivity(intent);
+    }
+
+    private void onClickUpdateUrl() {
+        String url = inputServerUrl.getEditText().getText().toString();
+        if (url.isEmpty()) {
+            inputServerUrl.setError("Server URL cannot be empty!");
+            return;
+        }
+        if (!url.startsWith("https://")) {
+            url = "https://" + url;
+        }
```

Comment:
Is it necessary that it will always start with `https`? Not sure probably you can ask Waylon if you're also not sure.